pkg/apis/v1alpha1/nodeoverlay.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NodeOverlaySpec struct {
+	// Requirements constrain when this NodeOverlay is applied during scheduling simulation.
+	// These requirements can match:
+	// - Well-known labels (e.g., node.kubernetes.io/instance-type, karpenter.sh/nodepool)
+	// - Custom labels from NodePool's spec.template.labels
+	// +kubebuilder:validation:XValidation:message="requirements with operator 'In' must have a value defined",rule="self.all(x, x.operator == 'In' ? x.values.size() != 0 : true)"
+	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt' or 'Lt' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt') ? (x.values.size() == 1 && int(x.values[0]) >= 0) : true)"
+	// +kubebuilder:validation:MaxItems:=100
+	// +required
+	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty"`
+	// PriceAdjustment specifies the price change for matching instance types. Accepts either:
+	// - A fixed price modifier (e.g., -0.5, 1.2)
+	// - A percentage modifier (e.g., +10% for increase, -15% for decrees)
+	// +kubebuilder:validation:Pattern=`^[+-](?:\d+(?:\.\d*)?|\.\d+)(?:%)?$`
+	// +optional
+	PriceAdjustment *string `json:"priceAdjustment,omitempty"`
+	// Price specifies amount for an instance types that match the specified labels. Users can override prices using a signed float representing the price override
+	// +kubebuilder:validation:Pattern=`^\d+(\.\d+)?$`
+	// +optional
+	Price *string `json:"price,omitempty"`
+	// Capacity adds extended resources only, and not replace any existing resources.
+	// These extended resources are appended to the node's existing resource list.
+	// Note: This field does not modify or override standard resources like CPU, memory, ephemeral-storage, or pods.
+	// +kubebuilder:validation:XValidation:message="invalid resource restricted",rule="self.all(x, !(x in ['cpu', 'memory', 'ephemeral-storage', 'pods']))"
+	// +optional
+	Capacity v1.ResourceList `json:"capacity,omitempty"`
```

Comment:
I'm not sure capacity appropriately covers this? Is this `CapacityOverrides`?

Reply:
> I'm not sure capacity appropriately covers this?

The resource here will map directly to capacity notion on nodes directly and not necessarily allocatable. To my I still think this is the best naming. Can you share you feel that this might not be the best name for it?


>  Is this CapacityOverrides?

These are not really overrides, as users are adding resources that are not on the nodes. I hesitate to insinuate any notion of overrides for capacity just because we have explicit validation against that